src/platform/backends/hyperv/hyperv_virtual_machine.cpp

```diff
@@ -453,6 +453,21 @@ void mp::HyperVVirtualMachine::ensure_vm_is_running()
 
 void mp::HyperVVirtualMachine::update_state()
 {
+    // Invalidate the management IP address on state update.
+    {
+        // The reason "why" is as follows:
+        // - The daemon starts and constructs the VM
+        // - The mgmt IP address gets cached on the first invocation of management_ipv4()
+        // - `multipass restart` initiates a guest reboot with "sudo reboot"
+        // -  It's an "out-of-band" reboot in perspective of Hyper-V backend since
+        //    it does not trigger the "on_restart" function afterwards.
+        // - The cached "vm->ip address is not reset
+        // - management_ipv4() continues to report the stale IP address from the previous boot
```

Comment:
I find this is a bit long, and I believe the top comment is pretty clear in the code. How about this instead?

```suggestion
    {
        // Cached IPs become stale when the guest is restarted from within. By resetting them here
        // we at least cover multipass's restart initiatives, which include state updates.
```

This makes it clear what cases this code helps with.